docs/developer-docs/security/security-best-practices/dos.mdx

```diff
@@ -4,32 +4,61 @@ keywords: [security, concept, denial of service, dos, cycles, captcha]
+* **Monitor cycles usage**: Regularly track your canisters cycles consumption and set alerts for any sudden spikes that may indicate an attack.
+* **Ingress message charging**: While charging for ingress messages (external requests to the canister) is not natively supported, custom solutions could be implemented to make sure that any expensive actions have costs associated with them.
+* **Filter ingress messages using inspect message**:  Certain non-critical checks can be placed in the inspect message function to filter out ingress update messages before they are executed by all nodes of a subnet. Since this code only runs on a single node, the execution does not consume cycles but it also shouldn't be relied upon for security critical checks such as access control. However, they can efficiently reject certain ingress messages early. Read the corresponding [documentation](https://internetcomputer.org/docs/current/references/ic-interface-spec#system-api-inspect-message) and [security best practice](https://internetcomputer.org/docs/current/developer-docs/security/security-best-practices/iam#do-not-rely-on-ingress-message-inspection) carefully for the caveats.
+
+## Protect Against Noisy Neighbors
+
+### Security Concern
+
+In a shared resource environment like the Internet Computer, multiple canisters can run on the same subnet. If one canister consumes too many resources (CPU, memory, etc.), it can negatively impact the performance of others on the same subnet. This is known as the "noisy neighbor" problem.
 
-* For query calls that cause significant computation and don't modify the state, it is advisable to not execute the expensive computation if the method is called as an update. However, keep in mind that query calls [don't provide authenticity guarantees](/docs/current/developer-docs/security/security-best-practices/data-integrity-and-authenticity#certify-query-responses-if-they-are-relevant-for-security), so this is a trade-off. Unfortunately, the execution mode of the query (whether it was called as query or update) is currently not directly exposed to the user code. However, one can call a method such as `ic0.data_certificate_present()` which returns `1` when called as query, and `0` for update methods. See the Interface Specification [section on certified data](/docs/current/references/ic-interface-spec#system-api-certified-data).
+### Recommendation
+
+To mitigate the "noisy neighbor" issue, manage your canister's resource allocation effectively:
+* **Memory allocation**: Memory can be reserved per canister by setting `memory_allocation`. This ensures that your canister can always allocate memory up to the requested `memory_allocation`. Thereby preventing other canisters from using up the subnets available memory. Note that this also reduces the upper bound of memory the canister can allocate to the same value. Monitoring actual memory usage against this value is important to avoid availability issues.
```

Comment:
```suggestion
* **Memory allocation**: Memory can be reserved per canister by setting `memory_allocation`, ensuring that your canister can always allocate memory up to the requested `memory_allocation` and preventing other canisters from using up the subnets available memory. Note that this also reduces the upper bound of memory the canister can allocate to the same value. Monitoring actual memory usage against this value is important to avoid availability issues.
```